### PR TITLE
 [MSINTEROP-720] Add functionality to parse products file

### PR DESCRIPTION
**Pull request by calebevans**
- Change the use of the word "service" to "product" per feedback in weekly meeting on 03/20
- Adds a class method `from_file` to the `Database.Products` class that is used to parse the "products" file and ensure the products defined in the file are populated in the database.
- Adds `verify_queries` to ensure only `blocker` and `critical-blocker` queries will be defined by users. New unit tests for this are added.
- Adds documentation for populating the "products" file.

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>


</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: qe_metrics/libs/database.py [hunk 1]
```diff
@@ -66,22 +67,48 @@ def bind_remote_db_connection(self) -> None:
...
+    class Products(DB_CONNECTION.Entity):  # type: ignore
         """
-        A class to represent the Services table in the database.
+        A class to represent the Products table in the database.
         """
 
         id = orm.PrimaryKey(int, auto=True)
         name = orm.Required(str, unique=True)
         jira_issues = orm.Set("JiraIssues")
+        queries = orm.Required(orm.Json)
```
Comment (myakove) [inline qe_metrics/libs/database.py:78]: `queries`  shouldn't be in the DB, only in the config file.

**Review by myakove — CHANGES_REQUESTED** [review]

**Review by rnetser — APPROVED** [review]

**Pull request merged**